vespajlib/src/main/java/com/yahoo/text/Text.java

```diff
@@ -50,9 +50,10 @@ public static boolean isTextCharacter(int codepoint) {
 
         return (codepoint < 0x80)
                 ? allowedAsciiChars[codepoint]
-                : isTextCharAboveUsAscii(codepoint);
+                : (codepoint <  Character.MIN_SURROGATE) || isTextCharAboveUsAscii(codepoint);
     }
     private static boolean isTextCharAboveUsAscii(int codepoint) {
+        if (codepoint <= Character.MAX_HIGH_SURROGATE) return false;
```

Comment:
That should be >=, but it's implicit, when the last if clause (below) has failed?